javascript/javascript2/week2/hyfBay/main.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector("ul");
+const sortType = document.getElementById("sort-selection");
+//dispaly products
+function renderProducts(products) {
+  // clear list before displaying new list
+  clearList();
+
+  //create list of countries
+  products.forEach((product) => {
+    let shipsTo = "";
+    product.shipsTo.forEach((country) => (shipsTo += `<li>${country}</li>`));
+
+    const li = document.createElement("li");
+    li.className = "prduct-display";
+    li.innerHTML = `<ul>
+    <li>${product.name}</li>
+    <li> ${product.price}</li> 
+    <li> ${product.rating}</li>
+    <li> <ul>${shipsTo}</ul></li>
+    </ul>`;
+    productsUl.appendChild(li);
+  });
+}
+// display whole list first time
+renderProducts(getSorted(products));
+
+//display products matches to names intered by user
+const searchInputField = document.getElementById("serach-input");
+
+searchInputField.addEventListener("keyup", () =>
+  renderProducts(getSorted(getSearchProductByName(products)))
```

Comment:
You call getSorted with no argument but it still works. Be aware of stuff like that!

Reply:
First of all I would like to say thank you for reviewing my home work and giving valuable comments. 
May be I am wrong but what I find here is that I didn't call getSorted with out argument actually you suggested to do it, for easy to understand for others. 
getSorted doesn't work with no argument. If we call with no argument it will throw error main.js:89 Uncaught TypeError: Cannot read property 'sort' of undefined at getSorted (main.js:89) .

Reply:
SO getSorted should have a parameter that is the an array of products. When calling getSorted you then need to make sure you are calling it with an array of products

Reply:
I hope I get your point. Hop fully we can fix it like this? . 
 //if atleast two products are in arrray then call getSorted 
  if (productsbyName.length > 1) {
    const sortedProducts = getSorted(productsbyName);
    renderProducts(sortedProducts);
    //if there is only one product then just render the product
  } else if (productsbyName.length === 1) renderProducts(productsbyName);
  //if there is no product then just clear the list
  else clearList();

Reply:
This looks super good yes!

Reply:
thanks